### ISSUE TITLE
Upcoming Media Card is back ...with a nice surprise for you! 😊

### ISSUE THREAD

**Issue by mkanet** [boralyl/steam-wishlist#19]
Hi @boraly FYI, [Upcoming Media Card](https://github.com/custom-cards/upcoming-media-card) is back with a bunch of new features!  

As a bonus, I released [Steam-Wishlist-UMC](https://github.com/mkanet/steam-wishlist-umc) integration.  It's based on your original Steam-Wishlist integration; except it leverages Upcoming Media Card's latest features _(such as clickable hyperlinks, collapse filter, sorting, general filter, etc)_.  It can do everything the original integration can, plus a lot more.  I also did the same for [Epic Games UMC](https://github.com/mkanet/epic_games_umc) integration for consistency.

**Upcoming Media Card**&nbsp;&nbsp;_(with Steam Wishlist UMC)_*&nbsp;&nbsp;*&nbsp;&nbsp;

Click **Play** to see it in action...

![Animated example](https://github.com/mkanet/steam-wishlist-umc/blob/main/assets/Steam-Wishlist-UMC-vs-Original-Steam-Wishlist.gif?raw=true)

**Note:**  I added a new setting (below) for Steam-Wishlist-UMC to allow displaying the rest of the wishlist items _(since non-sale items can now be conveniently collapsed/expanded as an option)_.
![show-non-saletoo](https://github.com/boralyl/steam-wishlist/assets/811008/4d4ef350-272a-48fd-a5a1-5c614d53465f)

PS:  I still haven't announced the return of Upcoming Media Card yet on the Home Assistant forum.  I plan to do this after doing some more testing.  Thank you for making the Steam Wishlist integration.  It's one of my new favorites!

**Comment by boralyl** [comment]
This looks great!  I

nstead of creating an entirely new integrations users will have to find, could you create a PR to add the required changes to this repo, so existing users won't have to use a different integration?  This also means there aren't multiple versions that need to be maintained.

**Comment by mkanet** [comment]
I would love to do a PR; in fact, that is what I wanted to do originally.  However, I wasn't able to adapt the original Steam Wishlist card to support Upcoming Media Card without making significant changes to UMC _just_ to be able to support and display sales wishlists specifically.  It really made a mess of things when I attempted to do that.  I even tried to make a special experimental version with had both sensors; but, that got way too complicated.

I know this might be too much to ask; but, any chance you would want to migrate to this newer UMC version?  Do you think existing Steam Wishlist users (including yourself) be willing to switch over?  If so, we can work together to coordinate this.  

BTW: Currently, the original Steam Wishlist integration and Steam Wishlist UMC integrations can run side-by-side without any conflicts to make it easier to migrate over; if that's what you want.  Just let me know.

**Comment by boralyl** [comment]
I'd gladly help get your PR working.  For what it's worth this integration isn't related to UMC, it just happens to sort of work with it if you use an old fork I created for the [nintendo wishlist card](https://github.com/custom-cards/nintendo-wishlist-card).

I took a quick look and it doesn't appear like it would be difficult to update.  From what I can tell comparing the states generated by each, each individual entity has some extra attributes.  Those would be easy to add without any breaking changes to existing users.  The sensor with games on sale has it's data slightly different, but I don't think it would be difficult to update.  It's currently stored in attributes as:

```json
{
  "on_sale": [...]
}
```

Whereas you are storing extra data for the card in a data key:

```json
{
  "data": [...]
}
```

It would be easy to merge the two so that things continue to work as before but would also allow for someone to use your card by just merging the data into the format:

```json
{
  "on_sale": [...],
  "data": [...],
}
```

Let me know if that doesn't make sense.

**Comment by mkanet** [comment]
I'm totally embarrassed. 🫢 I didnt realize I could just add both `on_sale` and `data`  to the same sensor's attributes.  Believe it or not, I never even tried.  It literally took me less time to do this than it did to write back to you :)

```yaml
    @property
    def extra_state_attributes(self):
        placeholders = {
            'title_default': '$title',
            'line1_default': '$rating',
            'line2_default': '$price',
            'line3_default': '$release',
            'line4_default': '$genres',
            'icon': 'mdi:arrow-down-bold',
        }
        games_on_sale = [game for game in self.games if game["sale_price"]]
        data_list_with_placeholders = [placeholders] + games_on_sale
        return {"on_sale": games_on_sale, "data": data_list_with_placeholders}
```

Its all working now.  The only issue (which isnt too bad) is I need to rename all the components back to their original names:  For example, `sensor.steam_wishlist_umc_xxxxxxxxxx`   `binary_sensor.steam_wishlist_umc_xxxxxxxxxxx`,  integration domain name, etc.  Once I do that, Ill submit my PR.  I might not get to it until next weekend though. 

Thank you so much for making this even possible.

![image](https://github.com/boralyl/steam-wishlist/assets/811008/e7222cb3-5545-49c4-b506-ec4d3664bc2a)

**Comment by mkanet** [comment]
Hi @boralyl I completed making the changes necessary to support Upcoming Media Card.  I also updated README.md with the respective instructions.  I will submit my PR soon.  I hope you are okay with the changes I made.  Let me know in case you need me to change something.

**Comment by boralyl** [comment]
Thanks for the PR!  Released in [v2.3.0](https://github.com/boralyl/steam-wishlist/releases/tag/v2.3.0)
